### ISSUE TITLE
File::Create does not create a new file on Windows if no data is written

### ISSUE THREAD

**Issue by Deedasmi** [rust-lang/rust#37953]
rustc 1.13.0 (2c6933acc 2016-11-07)

Minimal example:
```
use std::fs::File;

fn main() {
    let _ = File::create("test");
}
```

Expected behavior:
A file "test" is created in the local directory with 0 bytes. If there is an existing file named "test", it is truncated to 0 bytes.

Linux behavior:
As Expected

Windows 10 (64 bit gnu toolchain) behavior:
Program executes successfully. If the file already existed, it is truncated to 0 bytes as expected. However, if the file did not exist it is not created.

tl;dr: Windows doesn't 'touch' files.

**Comment by Deedasmi** [comment]
Sorry, user error. Wasn't running the code I thought I was.
